frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -124,11 +134,13 @@ def process_exception(self, request, exception, spider):
     def open(self, spider):
         log.msg('Starting frontier', log.INFO)
         if not self.frontier.manager.auto_start:
-            self.frontier.start()
+            scrapy_kwargs = {'spider': spider}
+            self.frontier.start(**scrapy_kwargs)
```

Comment:
There are many situations in which we need customization at spider level:

- in the case of get_producer_slot(), this was needed mainly because we already defined standard ways to assign requests to slots along our project, which has other components (i.e. external script that scan our db and schedules all urls from it with some metadata in HCF qdata). We were using a scrapy middleware for handling HCF before frontera existed, so migration to frontera required to have this adaptation in order to define a standard way to assign urls to slots along all the project and external script be coordinated with sitemap or directory spiders without generating duplicated requests (HCF storage deduplicates inside the same frontier slot)

- in case of cf_make_requests, this is unavoidable for many operations like controlling the creation of request parameters (headers, method, request metadata, etc.) and request callback from qdata and url. This is impossible to be decided only by the backend. Each spider has its own needs.

- We also need access to spider for configuring frontier parametrs using spider arguments, because they are dynamic (i.e. we cannot set it via a hardcoded setting), most importantly, from which slot to read, as we have several instances of a spider, each one reading from a different slot.